app/Http/Controllers/Auth/AdmissionController.php

```diff
@@ -165,54 +168,82 @@ public function updateNote(Request $request, Application $application): Redirect
-        //            foreach ($usersToDelete->get() as $user) {
-        //                if ($user->profilePicture!=null) {
-        //                    Storage::delete($user->profilePicture->path);
-        //                    $user->profilePicture()->delete();
-        //                }
-        //            }
-        //            $files = File::where('application_id', '!=', null);
-        //            foreach ($files->get() as $file) {
-        //                Storage::delete($file->path);
-        //            }
-        //            $files->delete();
-        //            Application::query()->delete();
-        //            $usersToDelete->forceDelete();
-        //
-        //            RoleUser::where('role_id', Role::get(Role::APPLICATION_COMMITTEE_MEMBER)->id)->delete();
-        //            RoleUser::where('role_id', Role::get(Role::AGGREGATED_APPLICATION_COMMITTEE_MEMBER)->id)->delete();
-        //        });
-        //
-        //        Cache::clear();
-        // return back()->with('message', 'Sikeresen jóváhagyta az elfogadott jelentkezőket');
-        return back()->with('error', 'Még nincs implementálva.');
+        $this->authorize('finalize', Application::class);
+        if(!$this->semester()) {
+            throw new \InvalidArgumentException('No semester can be retrieved from the application periodic event.');
+        }
+        DB::transaction(function () {
+            $admitted_applications = Application::query()->admitted()->get();
+            $not_admitted_applications = Application::query()->whereNotIn('id', $admitted_applications->pluck('id'))->get();
+            // admit users
+            foreach ($admitted_applications as $application) {
+                $application->user->update(['verified' => true]);
+                if($application->admitted_for_resident_status) {
+                    $application->user->setResident();
+                } else {
+                    $application->user->setExtern();
+                }
+                $application->user->workshops()->sync($application->admittedWorkshops);
+                $application->user->setStatusFor($this->semester(), SemesterStatus::ACTIVE);
+                $application->user->internetAccess->extendInternetAccess($this->semester()->getStartDate()->addMonth());
+            }
+            // delete data for not admitted users
+            $files = File::query()
+                ->whereIn('application_id', $not_admitted_applications->pluck('id')) // application files
+                ->orWhereIn('user_id', $not_admitted_applications->pluck('user_id')); // profile pictures
+            foreach ($files->get() as $file) {
+                Storage::delete($file->path);
+            }
+            $files->delete();
+            // soft deletes application, keep them for future reference
+            // (see https://github.com/EotvosCollegium/mars/issues/332#issuecomment-2014058021)
```

Comment:
@horcsinbalint

Reply:
I have not checked thoroughly, but this part LGTM. If it deletes all the data for rejected users but keeps the data for admitted users, it should be fine AFAIK (I think it is the behavior that we should aim to achieve).

One question is what happens in cases where a student start a `Jogorvoslati eljárás`. I would not want to go into details publicly about our backup strategy (as it is proprietary) but I think that should solve that issue.

Reply:
But the question about `Jogorvoslati eljárás` is not a real issue IMO.